modules/math/src/main/java/com/opengamma/strata/math/impl/statistics/descriptive/QuantileResult.java

```diff
@@ -0,0 +1,443 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.math.impl.statistics.descriptive;
+
+import java.io.Serializable;
+
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.PropertyDefinition;
+
+import com.opengamma.strata.collect.array.DoubleArray;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import org.joda.beans.Bean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+/**
+ * Object describing the result from a {@link QuantileCalculationMethod}.
+ */
+@BeanDefinition
+public final class QuantileResult
+  implements ImmutableBean, Serializable {
+  
+  /** The quantile value. */
+  @PropertyDefinition
+  private final double value;
+  /** The indices of sample data used in the calculation. For discrete methods, the length is 1 and for interpolated
+   * methods, the length is usually 2. */
+  @PropertyDefinition
+  private final int[] indices;
```

Comment:
Is it ever greater than 2? If not, maybe it is one `int` and one optional `int`.